### PR TITLE
Product policy chain management

### PR DESCRIPTION
**Pull request by eguzki**

### CODE REVIEW

File: doc/product-reference.md [hunk 1]
```diff
@@ -144,6 +145,17 @@ Specifies product method
...
+#### PolicyConfigSpec
+
+Specifies product policy config object
+
+| **Field** | **json field**| **Type** | **Info** | **Required** |
+| --- | --- | --- | --- | --- |
+| Name | `name` | string | Policy name | Yes |
+| Version | `version` | string | Policy version | Yes |
+| Enabled | `enabled` | boolean | Policy enabling switch | Yes |
+| Configuration | `configuration` | object | Policy configuration object | Yes. Minimum required is the empty object `{}` |
```
Comment (miguelsorianod) [inline doc/product-reference.md:157]: Is there some doc we can link to about the configuration spec format or something?

File: apis/capabilities/v1beta1/product_types.go [hunk 2]
```diff
@@ -648,6 +662,22 @@ func (d *ProductDeploymentSpec) AuthAppKey() *string {
...
+	Name string `json:"name"`
+
+	// Version defines the policy version
+	Version string `json:"version"`
+
+	// Configuration defines the policy configuration
+	// +kubebuilder:pruning:PreserveUnknownFields
+	Configuration runtime.RawExtension `json:"configuration"`
+
+	// Version defines the policy version
```
Comment (miguelsorianod) [inline apis/capabilities/v1beta1/product_types.go:677]: Incorrect comment

File: apis/capabilities/v1beta1/product_types.go [hunk 3]
```diff
@@ -59,6 +60,19 @@ const (
 	ProductFailedConditionType common.ConditionType = "Failed"
 )
 
+var (
+	// ApicastPolicy refers to the main functionality of APIcast to work with the 3scale API manager
+	// Needs to exist in the policy chain
+	ApicastPolicy = PolicyConfig{
```
Comment (miguelsorianod) [inline apis/capabilities/v1beta1/product_types.go:66]: Is this a global public variable for some reason? Do other packages need to access this?

File: apis/capabilities/v1beta1/product_types.go [hunk 4]
```diff
@@ -702,6 +732,10 @@ type ProductSpec struct {
 	// ProviderAccountRef references account provider credentials
 	// +optional
 	ProviderAccountRef *corev1.LocalObjectReference `json:"providerAccountRef,omitempty"`
+
+	// Policies holds the product's policy chain
+	// +optional
+	Policies []PolicyConfig `json:"policies,omitempty"`
```
Comment (miguelsorianod) [inline apis/capabilities/v1beta1/product_types.go:738]: Is PolicyConfig not being a pointer intentional?

File: doc/product-reference.md (hunk 1 above)
Comment (eguzki) [reply doc/product-reference.md:157]: AFAIK, there is not.

File: apis/capabilities/v1beta1/product_types.go (hunk 4 above)
Comment (eguzki) [reply apis/capabilities/v1beta1/product_types.go:738]: I see no reason to use pointers.

**Review by miguelsorianod — APPROVED** [review]

**Pull request merged**